packages/tools/viewer/src/viewerElement.ts

```diff
@@ -55,61 +52,49 @@ function parseColor(color: string | null | undefined): Nullable<Color4> {
     return new Color4(data[0] / 255, data[1] / 255, data[2] / 255, data[3] / 255);
 }
 
-export type HotSpot = ViewerHotSpotQuery & {
-    /**
-     * An optional camera pose to associate with the hotspot.
-     */
-    cameraOrbit?: [alpha: number, beta: number, radius: number];
-};
+type ResetMode = "auto" | "reframe" | [ResetFlag, ...flags: ResetFlag[]];
 
-/**
- * Generates a HotSpot from a camera by computing its spherical coordinates (alpha, beta, radius) relative to a target point.
- *
- * The target point is determined using the camera's forward ray:
- *   - If the ray intersects with a mesh in the model, the intersection point is used as the target.
- *   - If no intersection is found, a fallback target is calculated by projecting the distance
- *     between the camera and the model's center along the camera's forward direction.
- *
- * @param model The reference model used to determine the target point.
- * @param camera The camera from which the HotSpot is generated.
- * @returns A HotSpot object.
- */
-export async function CreateHotSpotFromCamera(model: Model, camera: Camera): Promise<HotSpot> {
-    await import("core/Culling/ray");
-    const scene = model.assetContainer.scene;
-    const ray = camera.getForwardRay(100, camera.getWorldMatrix(), camera.globalPosition); // Set starting point to camera global position
-    const camGlobalPos = camera.globalPosition.clone();
-
-    // Target
-    let radius: number = 0.0001; // Just to avoid division by zero
-    const targetPoint = Vector3.Zero();
-    const pickingInfo = scene.pickWithRay(ray, (mesh) => model.assetContainer.meshes.includes(mesh));
-    if (pickingInfo && pickingInfo.hit) {
-        targetPoint.copyFrom(pickingInfo.pickedPoint!); // Use intersection point as target
-    } else {
-        const worldBounds = model.getWorldBounds();
-        const centerArray = worldBounds ? worldBounds.center : [0, 0, 0];
-        const distancePoint = Vector3.FromArray(centerArray);
-        const direction = ray.direction.clone();
-        targetPoint.copyFrom(camGlobalPos);
-        radius = Vector3.Distance(camGlobalPos, distancePoint);
-        direction.scaleAndAddToRef(radius, targetPoint); // Compute fallback target
+function coerceEngineAttribute(value: string | null): ViewerElement["engine"] {

```

Comment:
Nit - should be KebabCase and not camelCase

Reply:
All of the non-exported module level functions in this package use camelCase. But since they are completely private, I could switch them to PascalCase. I thought we were only doing that for public/exported functions though. I did the same thing in SceneLoader.ts with the options related work.

Reply:
Will be weird to not stay consistent with the formatting of exported and non-exported functions.

Reply:
Ok I'm fine changing it, but let me do it in a follow up PR. This PR is already kinda big.